### ISSUE TITLE
Case-insensitive globbing does not work on case-sensitive file systems

### ISSUE THREAD

**Issue by davidmerfield** [mrmlnc/fast-glob#172]
### Environment

  * OS Version: macOS 10.12.6 
  * File system: Mac OS Extended (Case-sensitive, Journaled)
  * Node.js Version: v8.12.0
  * fast-glob Version: 2.2.6

### Actual behavior

Case-insensitive globbing does not work on case-sensitive file systems.

### Expected behavior

Case-insensitive globbing should work on case-sensitive file systems.

### Steps to reproduce

1. Create a case-sensitive volume on and mount it.
2. Install fast-glob on the case-sensitive volume.
3. Run the code sample on the case-sensitive volume.

### Code sample

```js
var fs = require("fs");
var fg = require("fast-glob");

fs.writeFileSync('A', "");

fg.sync('A'); 
// returns [ 'A' ]

// The documentation is ambiguous about how to enable 
// case-insensitive globbing. I include all four possible
// scenarios below since they all return an empty list and 
// at least one of them should not, per the docs.
fg.sync('a', { case: false});
fg.sync('a', { case: true});
fg.sync('a', { nocase: true});
fg.sync('a', { nocase: false});
// all return []
```

**Comment by mrmlnc** [comment]
This behaviour related only to static patterns (without glob expression, like `a.js` or `directory/a.js`). And… yeap, it's a bug.

**Comment by mrmlnc** [comment]
This issue will be fixed with release `fast-glob@2.2.7`. You can check release issue #180.
